samples/apps/copilot-chat-app/importdocument/Program.cs

```diff
@@ -120,11 +122,12 @@ private static async Task UploadFileAsync(FileInfo file, Config config, Guid cha
         }
         Console.WriteLine($"Successfully acquired User ID. Continuing...");
 
-        using var fileContent = new StreamContent(file.OpenRead());
-        using var formContent = new MultipartFormDataContent
+        using var formContent = new MultipartFormDataContent();
+        List<StreamContent> filesContent = files.Select(file => new StreamContent(file.OpenRead())).ToList();
```

Comment:
This approach seems a little memory inefficient as you open all files up front then iterate through them one at a time? 

An alternative to consider:
 
```suggestion
       foreach (var file in files)
       {
          formContent.Add(new StreamContent(file.OpenRead()), "formFiles", file.Name);

          //upload, then dispose
       }
```

Reply:
This is needed as we have to dispose the `StreamContent` objects at the end. 

The approach suggested will end up uploading one file at a time as opposed to all the files in a single request.

P.s. we cannot create the `StreamContent` in a for each loop as the for each loop will dispose the disposable objects coming out of the loop.